src/LEGO.AsyncAPI.Bindings/Sns/Statement.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using LEGO.AsyncAPI.Models;
+using LEGO.AsyncAPI.Models.Interfaces;
+using LEGO.AsyncAPI.Writers;
+
+namespace LEGO.AsyncAPI.Bindings.Sns;
+
+public class Statement : IAsyncApiElement
+{
+    public Effect Effect { get; set; }
+
+    /// <summary>
+    /// The AWS account or resource ARN that this statement applies to.
+    /// </summary>
+    // public StringOrStringList Principal { get; set; }
+    public StringOrStringList Principal { get; set; }
+
+    /// <summary>
+    /// The SNS permission being allowed or denied e.g. sns:Publish
+    /// </summary>
+    public StringOrStringList Action { get; set; }
+
+    public void Serialize(IAsyncApiWriter writer)
+    {
+        if (writer is null)
+        {
+            throw new ArgumentNullException(nameof(writer));
+        }
+
+        writer.WriteStartObject();
+        writer.WriteOptionalProperty(AsyncApiConstants.Effect, this.Effect.ToString());
+        writer.WriteOptionalObject(AsyncApiConstants.Principal, this.Principal, (w, t) => t.Serialize(w));
+        writer.WriteOptionalObject(AsyncApiConstants.Action, this.Action, (w, t) => t.Serialize(w));
+        writer.WriteEndObject();
+    }
+}
+
+public enum Effect
+{
+    Allow,
+    Deny
+}
+
+public class StringOrStringList : IAsyncApiElement
+{
+    public string StringValue { get; set; }
+
+    public List<string> StringList { get; set; }
+
+    public void Serialize(IAsyncApiWriter writer)
+    {
+        if (writer is null)
+        {
+            throw new ArgumentNullException(nameof(writer));
+        }
+
+        writer.WriteStartObject();
+        if (this.StringValue != null)
+        {
+            writer.WriteValue(this.StringValue);
```

Comment:
@VisualBean So, I used this to write the raw string when the value is not an array. However, if the value here is `someARN` for example, then the writer will write the string `someARN {}` adding the curly braces to the end. Same for `WriteRaw` I believe. I've looked at the functions but I can't work out why its adding the braces to the string when it writes it to YAML. Any ideas? 🥲